org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/settings/SettingsTest.java

```diff
@@ -130,12 +148,37 @@ public void formatSettings() {
 		// the InitializeParams
 		xmlFormattingOptions.merge(sharedXMLFormattingOptions);
 		Assert.assertEquals(5, xmlFormattingOptions.getTabSize()); // tab size is kept as 5 (and not updated with
-																   // shared value 10), because only the request's
-																   // formattingOptions object is allowed to define it.
-		Assert.assertFalse(xmlFormattingOptions.isInsertSpaces()); // insert spaces is kept as false because only the request's
-																   // formattingOptions object is allowed to define it.
+		// shared value 10), because only the request's
+		// formattingOptions object is allowed to define it.
+		Assert.assertFalse(xmlFormattingOptions.isInsertSpaces()); // insert spaces is kept as false because only the
+																																// request's
+		// formattingOptions object is allowed to define it.
 		Assert.assertTrue(xmlFormattingOptions.isJoinCommentLines());
 
 		Assert.assertTrue(xmlFormattingOptions.getQuotations().equals(XMLFormattingOptions.SINGLE_QUOTES_VALUE));
 	}
+
+	@Test
+	public void cachePathSettings() {
+		// Emulate InitializeParams#getInitializationOptions() object created as
+		// JSONObject when XMLLanguageServer#initialize(InitializeParams params) is
+		// called
+		
+		InitializeParams params = createInitializeParams(json);
+		Object initializationOptionsSettings = InitializationOptionsSettings.getSettings(params);
+		XMLLanguageServer languageServer = new XMLLanguageServer();
+
+		String originalUserHome = System.getProperty("user.home");
+		String userDir = System.getProperty("user.dir");
+		System.setProperty("user.home", userDir + "/" + targetTestFolder); // .../org.eclipse.lsp4xml/target/generated-test-sources/

```

Comment:
changing user.dir is not necessary here, could use anything under current target/ folder